### PR TITLE
Bluetooth: Audio: Media control configuration fixes

### PR DESCRIPTION
**Pull request by asbjornsabo**
Configuration fixes to enable the media control shell to run, and to not time out during GATT discovery.

### CODE REVIEW

**Review by Thalley — APPROVED** [review]
LGTM

File: tests/bluetooth/shell/audio.conf [hunk 1]
```diff
@@ -21,6 +21,10 @@ CONFIG_BT_HRS=y
 CONFIG_BT_WHITELIST=y
 CONFIG_BT_REMOTE_INFO=y
 CONFIG_BT_REMOTE_VERSION=y
+CONFIG_BT_CTLR_ADV_EXT=y
+CONFIG_BT_CTLR_ADV_PERIODIC=y
+CONFIG_BT_CTLR_SYNC_PERIODIC=y
+
```
Comment (cvinayak) [inline tests/bluetooth/shell/audio.conf:27]: Move this to board specific conf file, i.e. `tests/bluetooth/shell/boards/nrf52840dk_nrf52840.conf`

**Review by cvinayak — CHANGES_REQUESTED** [review]

File: tests/bluetooth/shell/audio.conf (hunk 1 above)
Comment (asbjornsabo) [reply tests/bluetooth/shell/audio.conf]: Done - for both 52840 and 5340

File: subsys/bluetooth/audio/Kconfig.mcs [hunk 2]
```diff
@@ -14,6 +14,7 @@ config BT_MCS
 	bool "Media Control Service Support"
 	select BT_CCID
 	select EXPERIMENTAL
+	depends on BT_L2CAP_TX_BUF_COUNT >= 10
```
Comment (cvinayak) [inline subsys/bluetooth/audio/Kconfig.mcs:17]: Instead add `default 10  if BT_MCS`  in `subsys/bluetooth/host/Kconfig.l2cap` under `BT_L2CAP_TX_BUF_COUNT`

**Review by cvinayak — CHANGES_REQUESTED** [review]

**Comment by asbjornsabo** [comment]
Changes since the first iteration of this PR
- move the controller configuration of advertising to board specific conf files
- change the L2CAP buffer setting from a "depends on" in the MCS config to a "deafult if MCS" in the l2cap conf file

File: subsys/bluetooth/host/Kconfig.l2cap [hunk 3]
```diff
@@ -9,6 +9,7 @@ config BT_L2CAP_TX_BUF_COUNT
 	int "Number of L2CAP TX buffers"
 	default NET_BUF_TX_COUNT if NET_L2_BT
 	default BT_BUF_ACL_TX_COUNT
+	default 10 if BT_MCS
```
Comment (cvinayak) [inline subsys/bluetooth/host/Kconfig.l2cap:12]: @tejlmand does the conditional default have to be in an order? like before the unconditional default value?

File: subsys/bluetooth/audio/Kconfig.mcs (hunk 2 above)
Comment (Thalley) [reply subsys/bluetooth/audio/Kconfig.mcs]: I disagree. If MCS require the BUF_COUNT to be 10 to work properly (which I guess is the case?), then `default` should not be used. `BT_L2CAP_TX_BUF_COUNT` can still be set to e.g. 2, and then you have a MCS that doesn't work as intended.

File: subsys/bluetooth/host/Kconfig.l2cap (hunk 3 above)
Comment (asbjornsabo) [reply subsys/bluetooth/host/Kconfig.l2cap:11]: Moved to before the unconditional default - in the above, the value was not set to 10.

**Review by cvinayak — APPROVED** [review]

File: subsys/bluetooth/audio/Kconfig.mcs (hunk 2 above)
Comment (cvinayak) [reply subsys/bluetooth/audio/Kconfig.mcs]: with `depends` `BT_MCS` is not available for users. Try using `ninja menuconfig` and you will not see MCS listed under audio services.

Comment (Thalley) [reply subsys/bluetooth/audio/Kconfig.mcs]: I see. I had not considered how `depends on` affect `ninja menuconfig` (I always just edit those in the text files instead...).

Comment (Thalley) [reply subsys/bluetooth/audio/Kconfig.mcs]: @asbjornsabo Should we add a BUILD_ASSERT for mcs.c to ensure that there's enough L2CAP buffers?

**Comment by cfriedt** [comment]
@jhedberg - good to merge?

**Review by Thalley — CHANGES_REQUESTED** [review]
If MCS cannot work with `BT_L2CAP_TX_BUF_COUNT < 10`, then we need to add a BUILD_ASSERT to ensure that.

File: subsys/bluetooth/audio/Kconfig.mcs (hunk 2 above)
Comment (asbjornsabo) [reply subsys/bluetooth/audio/Kconfig.mcs]: Added

**Review by Thalley — APPROVED** [review]

**Pull request merged**